core/src/main/java/org/kiwix/kiwixmobile/core/utils/DimenUtils.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.utils
+
+import android.app.Activity
+import android.content.Context
+import android.util.DisplayMetrics
+import android.util.TypedValue
+import androidx.appcompat.R
+
+object DimenUtils {
+  @JvmStatic fun Context.getToolbarHeight(): Int {
+    return resources.getDimensionPixelSize(TypedValue().apply {
+      theme.resolveAttribute(R.attr.actionBarSize, this, true)
+    }.resourceId)
+  }
+
+  @JvmStatic fun Activity.getWindowHeight(): Int =
+    computeDisplayMetric(this).heightPixels
+
+  @JvmStatic fun Activity.getWindowWidth(): Int =
+    computeDisplayMetric(this).widthPixels
+
+  private fun computeDisplayMetric(activity: Activity): DisplayMetrics =
```

Comment:
Why not convert this to extension function?

Reply:
I didn't prefer converting it to extension function because it used only in this file (private).

Reply:
Can be converted

Reply:
>I didn't prefer converting it to extension function because it used only in this file (private).

The reason for writing is extension functions is readability. Private functions should also be readable. I'd probably change it, maybe even change it to an extension property `computedDisplayMetrics`

so at the call site we now have 
```
  @JvmStatic fun Activity.getWindowWidth(): Int =    computedDisplayMetrics.widthPixels
```